### ISSUE TITLE
Bluetooth Gamepads Not Found In Linux

### ISSUE THREAD

**Issue by ezacharyk** [openfl/openfl#2507]
My  Bluetooth Switch Pro and Ouya controllers can connect to my Ubuntu 21.04 Linux computer and they are usable by Steam, but HaxeFlixel does not see them.

I tested this using the HaxeFlixel GamePad Test demo as well as the FlxAction method in the FlxAction and Mode demos. Only my wired controllers can be seen by HaxeFlixel.

I booted up my Windows partition and tested there, and both my Switch Pro and Ouya controllers were usable in both GamePad Test and FlxAction. So the issue lies in the Linux target.

Also, the HTML5 target could see my Ouya controller but not my Switch Pro controller in Windows and Linux.

I compiled the OpenFl sample: https://github.com/openfl/openfl-samples/tree/master/features/ui/GamepadInput
It showed my wired controllers but not my Bluetooth controllers.

Where would be a good place to start looking for a fix for this?

**Comment by joshtynjala** [comment]
It looks like GamePad mappings are defined in the constructor of Lime's `Window` class:

https://github.com/haxelime/lime/blob/c743fe10c404fb73003e24100f319a472cb4203f/src/lime/ui/Window.hx#L137

**Comment by ezacharyk** [comment]
@joshtynjala I tried adding my Gamepad mapping to that file and rebuilding Lime for Linux but it didn't work when I rebuilt the apps I used for testing. So I am still not sure what other steps would be necessary to add a controller mapping.

**Comment by Geokureli** [comment]
I don't have a linux machine but my bluetooth switch controller works on MacOS for me, on all the targets I've tried, does this work for you? https://haxeflixel.com/demos/GamepadTest/

I plan to get a linux machine soon, but until then I don't know how I would help

**Comment by ezacharyk** [comment]
@Geokureli Yeah, that works on HTML5 and Windows targets. But it doesn't see my Bluetooth controllers when compiled for Linux. Same for the OpenFl gamepad test. Good to know it works on Mac.
